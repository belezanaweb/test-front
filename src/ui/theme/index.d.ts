import 'styled-components'

import { ITheme } from './types'

declare module 'styled-components' {
  export interface DefaultTheme extends ITheme { }
}
