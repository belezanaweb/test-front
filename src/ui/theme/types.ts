export interface ITheme {
  font: string;
  colors: {
    base: string,
    baseAux: string,
    baseDark: string,
    baseLight: string,
    error: string,
    focus: string,
    primary: string,
    primaryLight: string,
    primaryDark: string,
    secondary: string,
    secondaryLight: string,
    secondaryDark: string
  };
}
