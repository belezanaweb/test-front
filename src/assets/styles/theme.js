export const defaultTheme = {
  borderRadius: '3px',

  fonts: {
    header: '13px',
    label: '12px',
    text: '13px',
    title: '14px',
    highlightText: '14px',
    input: '16px',
    button: '20px',
    desktop: '1rem'
  },

  colors: {
    primary: '#212122',
    secondary: '#999',
    orange: '#FF6C00',
    darkOrange: '#D45A00',
    ultraLightGray: '#E0E7EE',
    lightGray: '#CCC',
    mediumGray: '#E7E7E7',
    gray: '#EEE',
    lightBlack: '#212121',
    black: '#000',
    white: '#FFF',
    purple: '#A43287',
    red: '#F30'
  },

  shadow: '1px 1px 5px 0 rgba(0, 0, 29, 0.22)',

  breakpoints: {
    mobile: '0px',
    tablet: '576px',
    desktop: '1024px'
  }
}
