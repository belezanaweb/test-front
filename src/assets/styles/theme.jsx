export const colors = {
  background: '#eee',
  button: '#FF6C00',
  'button-hover': '#D45A00',
  'card-info-text': '#000',
  'font-color': '#212122',
  grey: '#999',
  'input-border': '#E7E7E7',
  'input-focus': '#A43287',
  'input-error': '#F30',
  orange: '#FF7800',
  placeholder: '#E0E7EE',
  'shadow-grey': '#CCC',
  white: '#fff'
}
