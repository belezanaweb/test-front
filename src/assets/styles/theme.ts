import { DefaultTheme } from 'styled-components'

const defaultTheme: DefaultTheme = {
  borderRadius: '3px',

  fonts: {
    label: '12px',
    header: '13px',
    text: '13px',
    title: '14px',
    highlightText: '14px',
    input: '16px',
    button: '20px'
  },

  colors: {
    primary: '#212122',
    secondary: '#999',
    accent: '#FF6C00',
    darkAccent: '#D45A00',
    ultraLightGray: '#E0E7EE',
    lightGray: '#CCC',
    mediumGray: '#E7E7E7',
    gray: '#EEE',
    lightBlack: '#212121',
    black: '#000',
    white: '#FFF',
    purple: '#A43287',
    red: '#F30'
  },

  shadows: {
    paper: '1px 1px 5px 0 rgba(0, 0, 29, 0.22)'
  },

  breakpoints: {
    mobile: '0px',
    tablet: '576px',
    deskop: '1024px'
  }
}

export { defaultTheme }