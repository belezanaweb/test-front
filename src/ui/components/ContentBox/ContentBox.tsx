import React, { FC } from 'react'

import * as S from './styled'

const ContentBox: FC = ({ children }) => <S.ContentBox>{children}</S.ContentBox>

export default ContentBox
