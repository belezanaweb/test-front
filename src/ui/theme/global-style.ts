import { createGlobalStyle } from 'styled-components'

const GlobalStyle = createGlobalStyle`
  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    font-family: ${({ theme }) => theme.font};
  }

  html {
    font-size: 62.5%;
  }

  body {
    background-color: ${({ theme }) => theme.colors.secondaryDark};
    padding-bottom: 8rem;
  }

  img {
    max-width: 100%;
    display: block;
  }

  li {
    list-style: none;
  }
`

export default GlobalStyle
