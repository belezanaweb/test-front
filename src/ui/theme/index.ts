export default {
  font: 'Helvetica Neue, Helvetica, Arial, sans-serif',
  colors: {
    base: '#CCCCCC',
    baseAux: '#999999',
    baseDark: '#212122',
    baseLight: '#FFFFFF',
    error: '#FF3300',
    focus: '#A43287',
    primary: '#FF6C00',
    primaryLight: '#FF7800',
    primaryDark: '#D45A00',
    secondary: '#E7E7E7',
    secondaryLight: '#E0E7EE',
    secondaryDark: '#EEEEEE'
  }
}
