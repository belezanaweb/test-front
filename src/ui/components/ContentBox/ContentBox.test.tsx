import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'

import theme from '../../theme'
import ContentBox from './'

describe('<ContentBox />', () => {
  it('should match a snapshot', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <ContentBox />
      </ThemeProvider>
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
