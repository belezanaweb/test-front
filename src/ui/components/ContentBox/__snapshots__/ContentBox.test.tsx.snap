// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContentBox /> should match a snapshot 1`] = `
.c0 {
  background-color: #FFFFFF;
  border-radius: 0.3rem;
  box-shadow: 0.1rem 0.1rem 0.5rem 0 rgba(0,0,29,0.22);
  padding: 1.2rem;
}

<section
  class="c0"
/>
`;
